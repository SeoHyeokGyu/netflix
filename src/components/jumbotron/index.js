import React from "react";

export default function Jumbotron(){
    return (
        <p>Jumba</p>
    );
}