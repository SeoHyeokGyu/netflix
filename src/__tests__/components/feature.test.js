import React from 'react';
import { getByText, render } from '@testing-library/react';
import { Feature } from '../../components';

describe('<Feature />', () => {
  it('renders the <Feature /> with populated data', () => {
    // eslint-disable-next-line no-shadow
    const { container, getByText } = render(
      <Feature>
        <Feature.Title>영화, TV 프로그램을 무제한으로.</Feature.Title>
        <Feature.SubTitle>다양한 디바이스에서 시청하세요. 언제든 해지하실 수 있습니다.</Feature.SubTitle>
      </Feature>
    );
    expect(getByText('영화, TV 프로그램을 무제한으로.')).toBeTruthy();
    expect(getByText('다양한 디바이스에서 시청하세요. 언제든 해지하실 수 있습니다.')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Feature /> with just a title', () => {
    // eslint-disable-next-line no-shadow
    const { container, queryByText, getByText } = render(
      <Feature>
        <Feature.Title>영화, TV 프로그램을 무제한으로.</Feature.Title>
      </Feature>
    );
    expect(getByText('영화, TV 프로그램을 무제한으로.')).toBeTruthy();
    expect(queryByText('다양한 디바이스에서 시청하세요. 언제든 해지하실 수 있습니다.')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Feature /> with just a sub-title', () => {
    // eslint-disable-next-line no-shadow
    const { container, queryByText, getByText } = render(
      <Feature>
        <Feature.SubTitle>다양한 디바이스에서 시청하세요. 언제든 해지하실 수 있습니다.</Feature.SubTitle>
      </Feature>
    );
    expect(getByText('다양한 디바이스에서 시청하세요. 언제든 해지하실 수 있습니다.')).toBeTruthy();
    expect(queryByText('영화, TV 프로그램을 무제한으로.')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
