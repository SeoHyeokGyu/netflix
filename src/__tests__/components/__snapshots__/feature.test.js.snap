// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders the <Feature /> with just a sub-title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cXQSCO"
>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 bjsrTz"
  >
    다양한 디바이스에서 시청하세요. 언제든 해지하실 수 있습니다.
  </h2>
</div>
`;

exports[`<Feature /> renders the <Feature /> with just a title 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cXQSCO"
>
  <h1
    class="feature__Title-sc-6swrp6-1 kGnPHh"
  >
    영화, TV 프로그램을 무제한으로.
  </h1>
</div>
`;

exports[`<Feature /> renders the <Feature /> with populated data 1`] = `
<div
  class="feature__Container-sc-6swrp6-0 cXQSCO"
>
  <h1
    class="feature__Title-sc-6swrp6-1 kGnPHh"
  >
    영화, TV 프로그램을 무제한으로.
  </h1>
  <h2
    class="feature__SubTitle-sc-6swrp6-2 bjsrTz"
  >
    다양한 디바이스에서 시청하세요. 언제든 해지하실 수 있습니다.
  </h2>
</div>
`;
